src/bsoncxx/exception/bsoncxx_error.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <bsoncxx/config/prelude.hpp>
+
+#include <limits>
+#include <system_error>
+
+namespace bsoncxx {
+BSONCXX_INLINE_NAMESPACE_BEGIN
+
+enum class bsoncxx_error : std::int32_t {
+    k_cannot_append_key_in_sub_array = 1,
+    k_cannot_close_array_in_sub_document,
+    k_cannot_close_document_in_sub_array,
+    k_cannot_perform_array_operation_on_document,
+    k_cannot_perform_document_operation_on_array,
+    k_need_key,
+    k_no_array_to_close,
+    k_no_document_to_close,
+    k_unmatched_key_in_builder,
+    k_unset_element,
+    k_wrong_element_type,
+    k_unknown = std::numeric_limits<std::int32_t>::max(),
+};
+
+///
+/// Translate a bsoncxx error into a std::error_code.
+///
+/// @param an error from bsoncxx
+/// @return an error_code
+///
+std::error_code make_error_code(bsoncxx_error error);
```

Comment:
Yeah, that's what I've got. make_error_code(mongocxx_error error) is in mongocxx/exception/mongocxx_error.cpp, and make_error_code(mongoc_error error) is in mongocxx/exception/private/mongoc_error.hpp.